saleor/core/utils/__init__.py

```diff
@@ -145,3 +148,10 @@ def prepare_unique_slug(slug: str, slug_values: Iterable):
         unique_slug = f"{slug}-{extension}"
 
     return unique_slug
+
+
+def prepare_unique_attribute_value_slug(attribute: "Attribute", slug: str):
+    value_slugs = attribute.values.filter(slug__startswith=slug).values_list(
+        "slug", flat=True
+    )
```

Comment:
This implementation performs additional database query for each slug. Are we sure that is the proper solution?

Reply:
Unlike previous implementation not for every but three specific types of attributes and now it's limited to a slug - not fetching every value. I will later see if it's possible to push it further but even now it's enormously faster than it was.